### DISCUSSION TITLE
Setup Redis using runtime configuration

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/nuxt#20195] [Questions] [answered]
I have Redis integration implemented as shown in the docs:
https://nuxt.com/docs/guide/directory-structure/server#example-using-redis

It works, only the connection is hard-coded in `nuxt.config.ts`. Is it possible to make it configurable using Nuxt runtime config variables?

I managed to this:
```
redis: {
  driver: 'redis',
  host: process.env.REDIS_HOST,
  port: process.env.REDIS_PORT,
  password: process.env.REDIS_PASS
}
```
but those are underlying system env variables, not the Nuxt ones. So it is an option, but not the ideal one.

I also tried `useRuntimeConfig()`, but got error it is undefined when used inside `nuxt.config.ts`

**Comment by maou-shonen** [comment]
related #18910

I have the same problem, Currently I have to use DNS to solve this.
the issue seems to be on [this line](https://github.com/unjs/nitro/blob/4da88f9f10928f8777f6f1fa04ddd1c0b600b9ed/src/rollup/plugins/storage.ts#L60).

**Comment by AloisSeckar** [comment]
Any chance someone has any idea other than using system env variables?

**Comment by pi0** [comment, marked as answer]
Please see https://github.com/unjs/nitro/issues/1161#issuecomment-1511444675 for the workaround.

**Reply by AloisSeckar** [reply]
Thank you, @pi0. This is exactly what I was looking for. I was thinking it could be resolved via a plugin, but was unable to figure out how to write it.
